Tests/test_cards.gd

```diff
@@ -15,12 +15,20 @@ func before_each():
 	
 	# fill deck with 50 default cards
 	_card_container.default_deck.resize(50)
-	_card_container.default_deck.fill(CardBase.new())
 	_card_container.max_hand_size = 10
 	_card_container.starting_hand_size = 5
 	
+	# assign cards names of Card1-Card50
+	for card_index: int in _card_container.default_deck.size():
```

Comment:
`in range(0,_card_container.default_deck.size())` ? will go from 0 to 49 so the index is the same as the card name
I don't know if it works without a range anyway

Reply:
I think 1-50 is a bit more clear personally

Reply:
Well that's a personal choice, but does it work with `_card_container.default_deck.size()` and not `range(_card_container.default_deck.size())` ?

Reply:
You mean does the current implementation properly loop through every index 0-49 without using range()? Yes it does I believe

Reply:
I'm curious, because 50 is an int, and `in` needs an iterator iirc

Reply:
Just double checked and yes, this syntax loops through all 0-49

Reply:
oh well that's strange I thought you needed a `range()` for it to work correctly.